### PR TITLE
2022 Page Weight Chapter

### LINKED ISSUE #2899: Page Weight 2022
# Page Weight 2022

Page Weight illustration

If you're interested in contributing to the Page Weight chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @fellowhuman1101 | @fellowhuman1101 @dwsmart | @CSteele-gh @mordy-oberstein | @drohe | - | @paulcalvano |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [ ] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [ ] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 Page Weight resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-page-weight on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1sEuLeW-o1N9B8WINy8iw5FblzSIkVOSedVu0JERrJVw/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/page-weight
[~google-sheets]: https://docs.google.com/spreadsheets/d/1JvJMiRsL6T9m_NEBHFh-rrQmU5a-ufdOKriSJbrEN8M/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/page-weight.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

### PR DESCRIPTION
**Pull request by tunetheweb**
Fixes #2899

The branch looks in pretty good shape (thanks @dwsmart and @fellowhuman1101 !) so going to try to edit in this PR and include in tomorrow's launch if I can. Might need to take a few liberties if authors not online.

Staging URL: https://20220925t170447-dot-webalmanac.uk.r.appspot.com/en/2022/page-weight
Home page quotes: https://20220925t170447-dot-webalmanac.uk.r.appspot.com/en/2022/?page-weight=fonts#featured-chapter

Assets:
- [Doc](https://docs.google.com/document/d/1sEuLeW-o1N9B8WINy8iw5FblzSIkVOSedVu0JERrJVw/edit#)
- [Sheet](https://docs.google.com/spreadsheets/d/1JvJMiRsL6T9m_NEBHFh-rrQmU5a-ufdOKriSJbrEN8M/edit#gid=1472139207)

### CODE REVIEW

**Review by dwsmart — APPROVED** [review]
Looks good to me!

File: src/content/en/2022/page-weight.md [hunk 1]
```diff
@@ -4,19 +4,443 @@ title: Page Weight
...
+
+In 2021, the 100th percentile of mobile sites used 20,452 kilobytes of font files. In 2022, these outliers swelled to 110 megabytes. This 540% growth was not seen in the year of year comparison for desktop which sat at 66,257 kilobytes in 2021 and 68,285 in 2022.
+
+However, the 100th percentile—while fun to investigate—will always show the worst of the web. At At the 90th percentile, the mobile font weight was a less extreme—though still large—401 kilobytes.
+
+More insights into the typographical nature of the web can be found in the [Fonts](./2021/third-parties) chapter.
+
+## Page weight by the numbers
+
+No we know what we are primarily interested in when considering page weight, let's dive into the details.
```
Comment (tunetheweb) [inline src/content/en/2022/page-weight.md:148]: I added this into text for this section, as gave Requests and Bytes their own sections.

File: src/content/en/2022/page-weight.md [hunk 2]
```diff
@@ -4,19 +4,443 @@ title: Page Weight
...
+  description="Box chart showing the distribution of image sizes by format. mp4 is the largest at the 50th percentile, with 534 KB, followed by f4v at 155 KB, flash at 257 KB, and flv at 720 KB.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vS-yCXlRH7TMaQg1T2YsESIiUNe4BSga5NWwEElHgU3ZaNLXNauwAxsn3k_el5v1fxvtc_YA5k9i689/pubchart?oid=1651375033&format=interactive",
+  sheets_gid="159262492",
+  sql_file="response_format_distribution.sql"
+  )
+}}
+
+## Adoption rates of byte-saving technologies
+
+So what can we do about all those bytes we're sending? Well obviously we could just stop sending them, but obviously they are being sent for a reason—you'd hope! So let's look at how to keep the content, but send it in more efficient ways than just stuffing kilobytes and kilobytes down the pipes.
```
Comment (tunetheweb) [inline src/content/en/2022/page-weight.md:338]: Added this as an intro to this section as we don't like "hanging headers" where we have an H2 immediately followed by an H3 with no text in between.

**Review by tunetheweb — COMMENTED** [review]
@fellowhuman1101 and @dwsmart I think this is good to merge now, but I did make some changes to what you had.

Primarily this was in reordering the content in the "Page Weight by the numbers" section as felt it was a jumping around a lot.

I also removed a lot of the (over) use of 100th percentile stats. As I commented in the doc:
> 100th percentiles are, by their very nature extreme. We use 10/25/50/75/90, however we include 100—but don't usually include in the chapter—so authors can see any "fun" stats. But shouldn't over do it on them and include the 100th percentile as something to regularly comment on.

Other than that it was a standard enough edit so hope I haven't upset you too much stomping all over your hard work! Was a great read btw, but hopefully even better now.

I'll also point out a couple of places I had to add some text. Feel free to reword.

Unfortunately there's not going to be a lot of time to review as we're releasing tomorrow (well today my time as just ticked over!), so will merge in about 10 hours or so to include in the release. I hope that's OK and you can, of course, follow up with any PRs for edits you disagree with!

Thanks again for your efforts and apologies only getting to the edit at the very last minute!

**Comment by tunetheweb** [comment]
Oh and I updated the staging link with the latest edits: https://20220925t170447-dot-webalmanac.uk.r.appspot.com/en/2022/page-weight

**Comment by dwsmart** [comment]
noticed a couple of little typos, pushed a fix. 

I'm happy with the edits you've done @tunetheweb, thank you for your hard, last minute work!

**Comment by tunetheweb** [comment]
And thank you for having done the conversion. Can’t tell you how happy I was to see that all done and in good shape when I got to this chapter! Don’t think it would have made it without that!

**Pull request merged**